src/main/java/org/ow2/proactive/connector/iaas/service/NodeCandidateService.java

```diff
@@ -45,11 +48,15 @@
     @Autowired
     private CloudManager cloudManager;
 
-    public Set<NodeCandidate> getNodeCandidate(String infrastructureId, String region, String imageReq) {
-        return Optional.ofNullable(infrastructureService.getInfrastructure(infrastructureId))
-                       .map(infra -> cloudManager.getNodeCandidate(infra, region, imageReq))
-                       .orElseThrow(() -> new NotFoundException("infrastructure id  : " + infrastructureId +
-                                                                " does not exists"));
+    public Map<String, Object> getNodeCandidate(String infrastructureId, String region, String imageReq, String token) {
```

Comment:
Could we create a class called PagedNodeCandidates or NodeCandidatesResponse to be the response type? And it could also be the return type for the CloudManager.getNodeCandidate to make it more clear than Pair.

Reply:
hmmm yes why not, nice idea!